lib/active_merchant/billing/gateways/datatrans.rb

```diff
@@ -0,0 +1,184 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class DatatransGateway < Gateway
+      self.test_url = 'https://api.sandbox.datatrans.com/v1/transactions/'
+      self.live_url = 'https://api.datatrans.com/v1/transactions/'
+
+      self.supported_countries = %w(CH GR US) # to confirm the countries supported.
+      self.default_currency = 'CHF'
+      self.currencies_without_fractions = %w(CHF EUR USD)
+      self.currencies_with_three_decimal_places = %w()
+      self.supported_cardtypes = %i[master visa american_express unionpay diners_club discover jcb maestro dankort]
+
+      self.money_format = :cents
+
+      self.homepage_url = 'https://www.datatrans.ch/'
+      self.display_name = 'Datatrans'
+
+      def initialize(options = {})
+        requires!(options, :merchant_id, :password)
+        @merchant_id, @password = options.values_at(:merchant_id, :password)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        authorize(money, payment, options.merge(auto_settle: true))
+      end
+
+      def authorize(money, payment, options = {})
+        post = add_payment_method(payment)
+        post[:refno] = options[:order_id].to_s if options[:order_id]
+        add_currency_amount(post, money, options)
+        add_billing_address(post, options)
+        post[:autoSettle] = options[:auto_settle] if options[:auto_settle]
+        commit('authorize', post)
+      end
+
+      def capture(money, authorization, options = {})
+        post = { refno: options[:order_id]&.to_s }
+        transaction_id, = authorization.split('|')
+        add_currency_amount(post, money, options)
+        commit('settle', post, { transaction_id: transaction_id, authorization: authorization })
+      end
+
+      def refund(money, authorization, options = {})
+        post = { refno: options[:order_id]&.to_s }
```

Comment:
because the order_id could be a number, this forces it to be a string (the sandbox doesn't hesitate about that, but documentation establishes that as a string). 

But considering it, it could be better to use fetch.